demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -706,341 +175,232 @@ int main(int argc, char* argv[]) {
+                persons_with_actions =
+                    custom::PersonDetActionRecPostProc::on(in, location, detect_confidences,
+                                                           priorboxes, action_con1,
+                                                           action_con2, action_con3,
+                                                           action_con4, ad_kernel_input);
+            }
+            cv::GOpaque<DrawingElements> draw_elements;
+            cv::GArray<TrackedObject> tracked_actions;
+            if (const_params.actions_type != TOP_K) {
+                /** Main demo scenario **/
+                cv::GOpaque<FaceTrack> face_track;
+                cv::GOpaque<size_t> work_num_frames;
+                /** Recognize actions and faces **/
+                std::tie(tracked_actions, face_track, work_num_frames)
+                    = custom::GetRecognitionResult::on(in, faces, persons_with_actions, embeddings, frec_kernel_input, const_params);
+
+                cv::GOpaque<std::string> stream_log, stat_log, det_log;
+                cv::GArray<std::string> face_ids(face_id_to_label_map);
+                /** Get roi and labels for drawing and set logs **/
+                std::tie(draw_elements, stream_log, stat_log, det_log)
+                    = custom::RecognizeResultPostProc::on(in, tracked_actions, face_track, face_ids, work_num_frames, const_params);
+                /** Main demo part of graph output **/
+                outs += GOut(work_num_frames, stream_log, stat_log, det_log);
+            } else {
+                /** Top action case **/
+                cv::GMat top_k;
+                /** Recognize actions **/
+                tracked_actions =
+                    custom::GetActionTopHandsDetectionResult::on(in, persons_with_actions);
+                 /** Get roi and labels for drawing **/
+                std::tie(draw_elements, top_k) = custom::RHands::on(in, tracked_actions, const_params);
+                /** Top action case part of graph output **/
+                outs += GOut(top_k);
+            }
+            /** Draw ROI and labels **/
+            auto rendered = cv::gapi::wip::draw::render3ch(frame,
+                                                           custom::BoxesAndLabels::on(frame, draw_elements, const_params));
+            /** Last graph output is frame to draw **/
+            outs += GOut(rendered);
+            return cv::GComputation(cv::GIn(in), std::move(outs));
+        });
+        auto networks = cv::gapi::networks(det_net, landm_net, reident_net, action_net);
+        cv::GStreamingCompiled cc = pp.compileStreaming(cv::compile_args(custom::kernels(),
+                                                        networks,
+                                                        TrackerParamsPack{ tracker_reid_params, tracker_action_params },
+                                                        logger_params));
```

Comment:
Probably this block should be indented deeper here, but it may change with the OMZ guidelines.